### PR TITLE
Changing csv_reader_py impl to return df from objmode

### LINKED ISSUE #1: Logistic regression example not working
Hi, I have an issue running the logistic regression example, when running it after installing h5py as recommended on documentation I get this import error:

```
Traceback (most recent call last):
  File "generate_data/gen_logistic_regression.py", line 7, in <module>
    @hpat.jit
  File "/home/alanm/miniconda3/envs/HPAT/lib/python3.6/site-packages/hpat-0.1.0-py3.6-linux-x86_64.egg/hpat/__init__.py", line 12, in jit
    from .compiler import add_hpat_stages
  File "/home/alanm/miniconda3/envs/HPAT/lib/python3.6/site-packages/hpat-0.1.0-py3.6-linux-x86_64.egg/hpat/compiler.py", line 5, in <module>
    from .hiframes import HiFrames
  File "/home/alanm/miniconda3/envs/HPAT/lib/python3.6/site-packages/hpat-0.1.0-py3.6-linux-x86_64.egg/hpat/hiframes.py", line 23, in <module>
    from hpat import pio
  File "/home/alanm/miniconda3/envs/HPAT/lib/python3.6/site-packages/hpat-0.1.0-py3.6-linux-x86_64.egg/hpat/pio.py", line 13, in <module>
    from hpat import pio_api, pio_lower, utils
  File "/home/alanm/miniconda3/envs/HPAT/lib/python3.6/site-packages/hpat-0.1.0-py3.6-linux-x86_64.egg/hpat/pio_lower.py", line 10, in <module>
    import hio
ModuleNotFoundError: No module named 'hio'
```

### PR DESCRIPTION
**Pull request by kozlov-alexey**
Motivation: returning Tuple of columns read from csv file with
pyarrow csv reader from objmode and further calling init_dataframe
ctor to create native DF turned out to be inefficient in sense of
LLVM IR size and compilation time. With this PR we now rely on DF
unboxing and return py DF from objmode.

**Compile time of read_csv + df.count():**
solutions\columns | 4 | 8 | 16 | 32 | 64 | 128 | 256
-- | -- | -- | -- | -- | -- | -- | --
Numba master + both SDC fixes   (2b8b0034d74) | 8.897234 | 9.306839 | 10.54691 | 12.52175 | 17.41399 | 30.47878 | 65.63396
Numba master + SDC fix #1 (964e498a9) | 9.283413 | 9.83861 | 13.30219 | 21.7165 | 53.07618 | 187.4615 | 1026.31
Numba 0.50.1  + SDC master | 9.212505 | 10.238 | 14.08183 | 25.16768 | 72.9872 | 290.3359 | 2141.832
Ratio (both fixes to master) | 1.035435 | 1.100051 | 1.335162 | 2.009917 | 4.191296 | 9.525835 | 32.63299

### CODE REVIEW

File: sdc/tests/test_io.py [hunk 1]
```diff
@@ -663,6 +663,8 @@ def test_impl():
 
         return test_impl
 
+    @skip_numba_jit("Lowering error while returning df from objmode:\n"
```
Comment (Hardcode84) [inline sdc/tests/test_io.py:666]: Are we going to do something with this?

Comment (kozlov-alexey) [reply sdc/tests/test_io.py:666]: @Hardcode84 Yes, the fix is rather simple - following lines are absent in sdc/hiframes/boxing.py:
```
                     elif ty_series == string_array_type:
                         native_val = unbox_str_series(string_array_type, series_obj, c)
+                    elif isinstance(ty_series, Categorical):
+                        native_val = unbox_Categorical(ty_series, arr_obj, c)
```
but I want to make it in a separate PR (add missed tests and move DF column unboxing into separate function). I'll make a new PR shortly.

Comment (Hardcode84) [reply sdc/tests/test_io.py:666]: ok

File: sdc/io/csv_ext.py [hunk 2]
```diff
@@ -541,84 +518,81 @@ def pandas_read_csv(
...
 
+    func_name = 'csv_reader_py'
+    return_columns = usecols if usecols and isinstance(usecols[0], str) else col_names
+
+    column_loc, _, _ = get_structure_maps(col_typs, return_columns)
+    df_type = DataFrameType(
+        tuple(col_typs),
+        types.none,
+        tuple(col_names),
+        column_loc=column_loc
```
Comment (kozlov-alexey) [inline sdc/io/csv_ext.py:531]: This implies has_parent=False (so we won't make reflection of changes made to native DF at boxing to python object), but it turns out that we don't support reflection for DFs at all. @AlexanderKalistratov, do you think it's a problem in this context (i.e. we should probably support reflection first)?

File: sdc/datatypes/hpat_pandas_functions.py [hunk 3]
```diff
@@ -255,45 +255,70 @@ def sdc_pandas_read_csv(
...
     else:
         return None
 
-    dtype_present = not isinstance(dtype, (types.Omitted, type(None)))
+    def _get_py_col_dtype(ctype):
+        """ Re-creates column dtype as python type to be used in read_csv call """
+        dtype = ctype.dtype
+        if ctype == string_array_type:
+            return str
+        if isinstance(ctype, Categorical):
```
Comment (kozlov-alexey) [inline sdc/datatypes/hpat_pandas_functions.py:309]: This can be improved with `_reconstruct_CategoricalDtype`

**Review by AlexanderKalistratov — APPROVED** [review]

**Pull request merged**